### ISSUE TITLE
MD030 false positive: bold text incorrectly flagged as unordered list markers

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#32]
## Bug Description

The MD030 rule (list-marker-space) is incorrectly flagging bold markdown syntax (`**text**`) as unordered list markers (`*`) with missing spaces.

## Expected Behavior

Bold text like `**Types**` should not be flagged as list marker violations.

## Actual Behavior

MD030 reports:
```
warning: MD030/list-marker-space: Unordered list marker spacing: expected 1 space(s) after '*', found 0
```

## Reproduction

Create a markdown file with:
```markdown
**Types**: feat, fix, docs
**Scopes**: cli, preprocessor, rules
```

Run `mdbook-lint lint file.md` and observe MD030 violations.

## Root Cause

The rule appears to be parsing `**` (bold syntax) as `*` (list marker) incorrectly.

## Related

- This affects any bold text that starts at the beginning of a line
- Similar issue could potentially affect italic text with `*text*`

## Environment

- mdbook-lint version: 0.1.0
- Found in: CLAUDE.md project documentation

**Comment by joshrotenberg** [comment]
This issue has been fixed in PR #35. The MD030 rule now properly distinguishes between emphasis syntax and list markers, preventing false positives with bold text.
